pkg/cloud/services/secretsmanager/secret.go

```diff
@@ -50,40 +58,57 @@ func (s *Service) Create(m *scope.MachineScope, data []byte) (string, error) {
 		Additional:  additionalTags,
 	})
 
-	name := s.secretName(m)
+	chunks := 1
+	buf := bytes.NewBuffer(data)
+	prefix := s.secretNamePrefix(m)
 
-	resp, err := s.scope.SecretsManager.CreateSecret(&secretsmanager.CreateSecretInput{
-		Name:         aws.String(name),
-		SecretBinary: data,
-		Tags:         converters.MapToSecretsManagerTags(tags),
-	})
+	for {
+		name := fmt.Sprintf("%s-%d", prefix, chunks-1)
```

Comment:
Would it be better to 0 index `chunks`, and do the increment after calling `buf.Next` if `len(chunk) > 0`? Could also return early in the case that `len(chunk) > 0`, since we wouldn't need to call `CreateSecret` in that case.

Reply:
The logic here is a currently a bit difficult to follow with chunks starting at 1 and the break below if `len(chunk) < maxSecretSizeBytes`

Reply:
I think the current logic would attempt to create a 0 byte secret in the case that the final chunk is exactly maxSecretSizeBytes

Reply:
sgtm